build_system/src/main.rs

```diff
@@ -26,16 +26,30 @@ macro_rules! arg_error {
 fn usage() {
     println!(
         "\
-Available commands for build_system:
+        Usage: build_system [command] [options]
 
-    cargo     : Run cargo command
-    clean     : Run clean command
-    prepare   : Run prepare command
-    build     : Run build command
-    test      : Run test command
-    info      : Run info command
-    clone-gcc : Run clone-gcc command
-    --help    : Show this message"
+    Available commands for build_system:
+
+        cargo     : Executes a cargo command. Use 'cargo --help' for a list of cargo commands.
+        clean     : Cleans the build directory, removing all compiled files and artifacts.
+        prepare   : Prepares the environment for building, including fetching dependencies and setting up configurations.
+        build     : Compiles the project. Use 'build --help' for build options.
+        test      : Runs tests for the project. Use 'test --help' for test options.
+        info      : Displays information about the build environment and project configuration.
+        clone-gcc : Clones the GCC compiler from a specified source. Use 'clone-gcc --help' for options.
+        --help    : Shows a help message.
+
+        Options:
+        -h, --help    : Displays this help message.
+
+        Examples:
+        build_system build
+        build_system test --release
+        build_system info
+        build_system clone-gcc --source=https://example.com/gcc.tar.gz
+
+        Note: Replace 'build_system' with the actual name of your build script executable if different.
```

Comment:
```suggestion
Examples:
        ./y.sh build
        ./y.sh test --release
        ./y.sh info
        ./y.sh clone-gcc --source=https://example.com/gcc.tar.gz
```

We use the [`y.sh` script](https://github.com/rust-lang/rustc_codegen_gcc/blob/master/y.sh) to access the build system.